### ISSUE TITLE
add `no-optional-and-default-together` rule

### ISSUE THREAD

**Issue by ulrichstark** [marcalexiei/eslint-plugin-zod#52]
### 🧩 Description

If some zod type already has `.default(SOME_VALUE)`, chaining `.optional()` is redundant because having a default value implies that the type can be optional/undefined. Reporting these cases would result in less code and slightly smaller and easier schemas.

### ⚙️ Options

N/A

### ✅ Valid Code Examples

```typescript
z.string().default("Hello World")
```

### ❌ Invalid Code Examples

```typescript
z.string().optional().default("Hello World")
```

**Comment by marcalexiei** [comment]
Consider the following scenarios:

- In one case a user might want to remove optional because is making the schema stricter
- On the contrary another might want to the schema more loose adding a default and removing an optional.

Based on this what do you think about the following rule:

## `no-optional-and-default-together`

Reports whenever both `.optional()` and `.default()` appear in the same Zod chain, regardless of order.

### Options

- Name: `preferredMethod` 
- Type: `string` 
- Default: `none` 
- Values: 
   - `none` 
   - `default`
    - `optional`

#### `preferredMethod: none`

Reports the error, no fix provided.

#### `preferredMethod: default `

Reports the error, provides a fix to remove `optional`.

#### `preferredMethod: optional `

Reports the error, provides a fix to remove `default`.

**Comment by marcalexiei** [comment]
No problem, thanks for sharing your idea!

I'll find some time to implement it next week 🚀
